trailblazer/services/job_service/job_service.py

```diff
@@ -48,6 +49,7 @@ def update_jobs(self, analysis_id: int) -> None:
                 self.store.update_tower_run_status(analysis_id)
         except Exception as error:
             LOG.error(f"Failed to update jobs {analysis.case_id} - {analysis.id}: {error}")
+            raise JobServiceError from error
```

Comment:
How does this help in preventing the problematic behaviour? Will they show up as "failed" instead of "cancelled"?

Reply:
The old logic will ensure the analysis status is set to `ERROR` when an exception bubbles up. The problem was that the exception was not propagated from the job service.